pyvista/core/utilities/geometric_objects.py

```diff
@@ -363,6 +364,353 @@ def Sphere(
     return surf
 
 
+def SolidSphere(
+    outer_radius=0.5,
+    inner_radius=0.0,
+    radius_resolution=5,
+    start_theta=0.0,
+    end_theta=360.0,
+    theta_resolution=30,
+    start_phi=0.0,
+    end_phi=180.0,
+    phi_resolution=30,
+    radius=None,
+    theta=None,
+    phi=None,
+    center=(0.0, 0.0, 0.0),
+    direction=(0.0, 0.0, 1.0),
+):
+    """Create a solid sphere.
+
+    Parameters
+    ----------
+    outer_radius : float, default: 0.5
+        Outer radius of sphere.
+
+    inner_radius : float, default: 0.0
+        Inner radius of sphere.
+
+    radius_resolution : int, default: 5
+        Number of points in radial direction.
+
+    start_theta : float, default: 0.0
+        Starting longitude angle in degrees.
+
+    end_theta : float, default: 360.0
+        Ending longitude angle in degrees.
+
+    theta_resolution : int, default: 30
+        Number of points in theta direction.
+
+    start_phi : float, default: 0.0
+        Starting latitude angle in degrees.
+
+    end_phi : float, default: 180.0
+        Ending latitude angle in degrees.
+
+    phi_resolution : int, default: 30
+        Number of points in phi direction,
+        inclusive of polar axis if applicable.
+
+    radius : sequence[float], optional
+        A monotonically increasing sequence of values specifying radial
+        points. If specified, it overrides any choices for ``outer_radius``,
+        ``inner_radius``, and ``radius_resolution``.
+
+    theta : sequence[float], optional
+        A monotonically increasing sequence of values specifying theta
+        points in degrees. If specified, it overrides any choices for
+        ``start_theta``, ``end_theta``, and ``theta_resolution``.
+
+    phi : sequence[float], optional
+        A monotonically increasing sequence of values specifying phi
+        points in degrees. If specified, it overrides any choices for
+        ``start_phi``, ``end_phi``, and ``phi_resolution``.
```

Comment:
Also a typical pain in the ass, but there are many conventions for defining spherical coordinates. Later in the polar transformation you have
```py
        x = r * np.sin(phi) * np.cos(theta)
        y = r * np.sin(phi) * np.sin(theta)
        z = r * np.cos(phi)
```
Which is exactly not how we'd call these in physics. We always use `phi` for the azimuthal angle and `theta` for the polar angle, which would correspond to
```py
        x = r * np.sin(theta) * np.cos(phi)
        y = r * np.sin(theta) * np.sin(phi)
        z = r * np.cos(theta)
```
So I think it would be very helpful to unambiguously define what the angles mean.

Reply:
This is how it is defined for `Sphere`.  I had actually written it the way you defined first and then swapped the meaning of phi and theta in https://github.com/pyvista/pyvista/pull/4965/commits/e75c3ee4d336b2a4e03ce9f36971cca5ffea6847 to be consistent. I more or less copied the description of the parameters from `Sphere.  We have `latitude` and `longitude` currently, but we can use `azimuthal` and `polar`.

Reply:
Yeah, I much prefer the physics convention but consistency is more important than convention preference. I think if it's defined a certain way for `Sphere` it has to be the same way for `SolidSphere`. I think `scipy` spherical harmonics might use the `phi=latitude` convention as well. I think `theta` and `phi` names are fine to stick with as long as it is explained in the documentation (I think it is already explained somewhere) that `phi` is latitude (polar) and `theta` is longitude (azimuthal). No need to introduce new conventions like `latitude/polar` into the parameter names.

Reply:
Yup, already the wolfram page I linked in a different comment follows the same convention, so it's definitely valid, just needs to be clarified.

Reply:
We have a much more comprehensive discussion in another comment.  I'm going to resolve this one in the hope that the discussion converges there.